Adding tty logging to the useradd command

edneville opened the pull request:
This commit adds a from= field to the end of the useradd log entry.
Casting user_name to tallylog_reset to silence a compiler warning.

Closes #198.

hallyn commented:
Thanks.  The patch looks good, however please make two changes (or I can do it if you prefer):

1. squash the two commits
2. (if I'm not overlooking a reason why it's not possible) instead of casting user_name to char*, instead have tallylog_reset() take a const char *.

edneville commented:
Thanks for looking @hallyn. I've removed the cast and changed the function as suggested. I think you'll have to do a squash merge as I've pushed commits to the source branch.

hallyn commented:
Thanks!
